src/zone-patch/index.js

```diff
@@ -44,13 +44,68 @@ function wrapDescribeInZone(describeBody) {
 // Create a proxy zone in which to run `test` blocks so that the tests function
 // can retroactively install different zones.
 const testProxyZone = ambientZone.fork(new ProxyZoneSpec());
-function wrapTestInZone(testBody) {
+function wrapTestInZone(testBody, eachArgs) {
   if (testBody === undefined) {
     return;
   }
-  return function(...args) {
-    return testProxyZone.run(testBody, null, args);
-  };
+
+  if (!eachArgs || eachArgs.length === 0 || eachArgs[0].length === 0) {
+    // If we are not handling `test.each`, then the parameter of `testBody`
+    // will be 0 or 1, if it is 1, then we need to return a function with
+    // done parameter
+    return testBody.length === 0
+      ? () => testProxyZone.run(testBody, null)
+      : done => testProxyZone.run(testBody, null, [done]);
+  } else {
+    /*
+      If we are handling `test.each`, and the each arguments have data,
+      then we need to dynamically create a Function, the reason is we can
+      only know the parameters of `test.each` method at runtime
+      For example:
+      ```
+      test.each([[1, 2]])('test.each', (arg1, arg2) => {});
+      ```
+      In this case we need to return a function like this
+      ```
+      return function(arg1, arg2) {
+        return testProxyZone.run(testBody, null, [arg1, arg2]);
+      }
+      ```
+      The most important part is we have to explicitly return the function
+      with 2 parameters, we can not return something like this,
+      ```
+      return function() {
+        return testProxyZone.run(testBody, null, arguments);
+      }
+      ```
+      It looks the logic is the same, but the explicit one will tell jest
+      the number of the parameters, and jest can handle `done()` or without `done()`
+      correctly.
+
+      And also because we are using `new Function` syntax here, the function body can not
+      access local scope, so we can not use `testProxyZone` or `testBody` variable
+      directly, so we have to pass them as parameters.
+    */
```

Comment:
I really appreciate your detailed explanation, but I think we should put most of this in the PR itself.
Would be great if you could narrow it down to like 5-10 lines, you can reference the PR itself for more information.
And maybe also use single line comments with `//`.

Reply:
@wtho , sure, thanks, I have updated the comments here.